### PR TITLE
Refactor PHPUnit deprecated methods.

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ x ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Fix deprecations of a test framework. (PHPUnit).

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Check build is completed w/o errors.

### CODE REVIEW

**Review by kuzmany — DISMISSED** [review]
Good 👍

File: app/bundles/CoreBundle/Tests/Unit/DependencyInjection/Builder/Metadata/ConfigMetadataTest.php [hunk 1]
```diff
@@ -16,8 +16,18 @@ class ConfigMetadataTest extends TestCase
 
     protected function setUp(): void
     {
+        $fieldHelperReflection = new \ReflectionClass(BundleMetadata::class);
+
+        $methods = array_map(static function (\ReflectionMethod $m): string {
+            return $m->name;
+        }, $fieldHelperReflection->getMethods());
```
Comment (escopecz) [inline app/bundles/CoreBundle/Tests/Unit/DependencyInjection/Builder/Metadata/ConfigMetadataTest.php]:
```suggestion
        $methods = array_map(static fn (\ReflectionMethod $m): string => $m->name, $fieldHelperReflection->getMethods());
```
I kinda enjoy the arrow functions. But this is just a nitpick.

Comment (escopecz) [reply app/bundles/CoreBundle/Tests/Unit/DependencyInjection/Builder/Metadata/ConfigMetadataTest.php]: As I can see that this diff of method names is repeated multiple times in this PR, could you DRY it to some static helper perhaps?

**Comment by escopecz** [comment]
I changed the target branch from 5.1 to 5.x as we are getting back to our classic branching strategy where 5.x will be for features and 5.0 for bug fixes.

File: app/bundles/CoreBundle/Tests/Unit/DependencyInjection/Builder/Metadata/ConfigMetadataTest.php (hunk 1 above)
Comment (biozshock) [reply app/bundles/CoreBundle/Tests/Unit/DependencyInjection/Builder/Metadata/ConfigMetadataTest.php]: Refactored the code, so no reflection is needed.

**Review by escopecz — APPROVED** [review]
I see no issues. Do you think we'll be able to get to PHPUNIT 10 with these changes?

**Comment by biozshock** [comment]
@escopecz i doubt that i've changed all deprecated methods. But will see when i have some time.

**Pull request merged**